hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -636,6 +637,12 @@ protected Map<String, String> properties() {
     return catalogProperties == null ? ImmutableMap.of() : catalogProperties;
   }
 
+  @Override
+  public void close() throws IOException {
+    super.close();
+    fileIO.close();
```

Comment:
`fileIO` could be null at this point if initialization failed and close is called immediately after it, so it would be good to add a null check.

Generally speaking, there's no guarantee that adding `close()` here actually fixes the exception you mentioned in the PR. You would most likely need a similar concept to what was introduced by #7487 / #8315

Reply:
I just created a `fileIOCloser` to track the IO files and close them when the catalog is closed, could your recheck please?